### PR TITLE
Split test_h1_pool_strategy into LRU and MRU tests and relax constraints

### LINKED ISSUE #769: Improve stability of test_h1_pool_strategy integration test
test_h1_pool_strategy is flakey and fails due to unrelated changes.

Examples:
https://app.circleci.com/pipelines/github/envoyproxy/nighthawk/3202/workflows/48283cb9-ae75-47f7-9738-77547c8067b2/jobs/30000
https://app.circleci.com/pipelines/github/envoyproxy/nighthawk/3190/workflows/46af4767-da13-4b3c-9b33-026d6f26d2b8/jobs/29865
https://app.circleci.com/pipelines/github/envoyproxy/nighthawk/3189/workflows/0596aea3-3b34-4fad-b55d-9e7731ae5460/jobs/29846

### PR DESCRIPTION
**Pull request by tomjzzhang**
Splits test_h1_pool_strategy integration test into two separate tests for better granularity. 

Relax the constraints of each test to be more resilient to a variation of requests sent while still verifying the behavior of the connection pool strategies. 

Increased the duration of each test to also improve the overall stability of traffic sent by nighthawk. 

closes #769 

Signed-off-by: tomjzzhang <4367421+tomjzzhang@users.noreply.github.com>

### CODE REVIEW

File: test/integration/test_connection_management.py [hunk 1]
```diff
@@ -102,36 +106,56 @@ def test_http_h2_connection_management_single_request_per_conn_1(http_test_serve
   _connection_management_test_request_per_connection(http_test_server_fixture, 5, True)
 
 
-def test_h1_pool_strategy(http_test_server_fixture):
+def test_h1_pool_strategy_mru(http_test_server_fixture):
   """Test connection re-use strategies of the http 1 connection pool.
 
   Test that with the "mru" strategy only the first created connection gets to send requests.
```
Comment (dubious90) [inline test/integration/test_connection_management.py:112]: preexisting but can we define what mru is here in this function comment.  (and in its method, define lru)

File: test/integration/test_connection_management.py [hunk 2]
```diff
@@ -102,36 +106,56 @@ def test_http_h2_connection_management_single_request_per_conn_1(http_test_serve
...
-
   _, logs = http_test_server_fixture.runNighthawkClient([
-      "--rps 5", "-v", "trace", "--connections", "2", "--prefetch-connections",
+      "--rps 5", "-v", "trace", "--duration", "20", "--connections", "2", "--prefetch-connections",
       "--experimental-h1-connection-reuse-strategy", "mru", "--termination-predicate",
       "benchmark.http_2xx:4",
       http_test_server_fixture.getTestServerRootUri()
   ])
 
+  # Expect the second connection to send any messages
```
Comment (dubious90) [inline test/integration/test_connection_management.py]: typo: to not send

File: test/integration/test_connection_management.py (hunk 1 above)
Comment (tomjzzhang) [reply test/integration/test_connection_management.py]: Done.

File: test/integration/test_connection_management.py (hunk 2 above)
Comment (tomjzzhang) [reply test/integration/test_connection_management.py]: Done.

File: test/integration/test_connection_management.py [hunk 3]
```diff
@@ -38,6 +38,10 @@ def _run_with_number_of_connections(fixture,
   return counters
 
 
+def _count_log_lines_with_substring(logs, substring):
```
Comment (mum4k) [inline test/integration/test_connection_management.py:41]: This likely deserves to be a function in a separate library with its own unit test.

We could add it to one of the existing helper libraries, or define a new one. The existing unit tests are here:

https://github.com/envoyproxy/nighthawk/tree/main/test/integration/unit_tests

**Review by mum4k — CHANGES_REQUESTED** [review]
Looks good, only one comment about placement and test coverage for a test utility.

File: test/integration/test_connection_management.py (hunk 3 above)
Comment (tomjzzhang) [reply test/integration/test_connection_management.py]: Done.

File: test/integration/utility.py [hunk 4]
```diff
@@ -72,3 +72,16 @@ def get_execution_duration_from_global_result_json(global_result_json):
         "the execution_duration '{} doesn't end with the expected suffix 's' for seconds".format(
             duration_json_string))
   return float(duration_json_string.rstrip('s'))
+
+
+def count_log_lines_with_substring(logs, substring):
+  """Count the number of log lines containing the supplied substring.
+
+  Args:
+    logs: Nighthawk client log output.
```
Comment (mum4k) [inline test/integration/utility.py:81]: Since we aren't using pytype, we also need to indicate the object type of each argument and return value.

Args:
  logs: A string, a log output to count the log lines in.

...

File: test/integration/unit_tests/test_utility.py [hunk 5]
```diff
@@ -40,5 +40,17 @@ def test_get_execution_duration_from_global_result_json_missing_suffix():
     utility.get_execution_duration_from_global_result_json(global_result_json)
 
 
+def test_count_log_lines_with_substring_counts_only_lines_with_substring():
```
Comment (mum4k) [inline test/integration/unit_tests/test_utility.py:43]: Thank you for adding the coverage, should we also add a test case with zero lines and one which has lines, but none of them matches?

Comment (tomjzzhang) [reply test/integration/unit_tests/test_utility.py:43]: Done.

File: test/integration/utility.py (hunk 4 above)
Comment (tomjzzhang) [reply test/integration/utility.py]: Done.

File: test/integration/unit_tests/test_utility.py [hunk 6]
```diff
@@ -52,5 +52,24 @@ def test_count_log_lines_with_substring_counts_only_lines_with_substring():
...
+  ERROR: log example 3 \r\n
+  INFO: log example 1 \r\n
+  """
+  count = utility.count_log_lines_with_substring(logs, "log example 4")
+  assert count == 0
+
+
+def test_count_log_lines_with_substring_returns_zero_with_empty_logs():
+  """Test that returns zero if there are no logs."""
+  logs = """"""
```
Comment (mum4k) [inline test/integration/unit_tests/test_utility.py:69]: (nit, optional) This can now be a regular empty string.

logs = ""

**Review by mum4k — APPROVED** [review]

File: test/integration/unit_tests/test_utility.py (hunk 6 above)
Comment (tomjzzhang) [reply test/integration/unit_tests/test_utility.py]: oops. Done.

**Review by mum4k — APPROVED** [review]
